tokio-console/src/config.rs

```diff
@@ -238,3 +313,67 @@ impl FromStr for RetainFor {
         }
     }
 }
+
+// === impl ColorToggles ===
+
+impl ColorToggles {
+    pub fn color_durations(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+
+    pub fn color_terminated(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+}
+
+// === impl ColorToggles ===
+
+impl ConfigFile {
+    fn from_config() -> Option<Self> {
+        let mut base = dirs::config_dir();
+        if let Some(path) = base.as_mut() {
+            path.push("tokio-console/console.toml");
+        }
+        let base = base.and_then(|path| fs::read_to_string(path).ok());
+        let base_file: Option<ConfigFile> = base.and_then(|raw| toml::from_str(&raw).ok());
```

Comment:
i think we could simplify this a bit by moving more of this code into the `if let`. here's my suggestion (assuming that we change this method to return a `Result`, as I suggested)
```suggestion
        let base_file = if let Some(path) = base.as_mut() {
            path.push("tokio-console/console.toml");
            fs::read_to_string(path).ok()
                .map(|raw| toml::from_str::<ConfigFile>(&raw))
                .transpose()?
        } else {
            None
        }
```